src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Extensions/ResolveFieldContextExtensions.cs

```diff
@@ -8,14 +7,15 @@ namespace VirtoCommerce.XDigitalCatalog.Extensions
 {
     public static class ResolveFieldContextExtensions
     {
-        public static T GetCatalogQuery<T>(this IResolveFieldContext context) where T: ICatalogQuery
+        public static T GetCatalogQuery<T>(this IResolveFieldContext context) where T : ICatalogQuery
         {
             var result = AbstractTypeFactory<T>.TryCreateInstance();
-            result.StoreId = context.GetArgument<string>("storeId") ?? context.GetValue<string>("storeId");
-            result.UserId = context.GetArgument<string>("userId") ?? context.GetValue<string>("userId");
-            result.CurrencyCode = context.GetArgument<string>("currencyCode") ?? context.GetValue<string>("currencyCode");
-            result.CultureName = context.GetArgument<string>("cultureName") ?? context.GetValue<string>("cultureName");           
-            return result;           
+            result.StoreId = context.GetArgumentOrValue<string>("storeId");
+            result.UserId = context.GetArgumentOrValue<string>("userId") ?? context.GetCurrentUserId();
```

Comment:
I think we should get the user id from the session if it doesn't  pass as an argument
https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs#L44